src/rect.js

```diff
@@ -78,10 +78,110 @@ var cornerstoneMath = (function (cornerstoneMath) {
         return minDistance;
     }
 
+    // Returns rectangle points
+    function rectToPoints (rect) {
+        var rectPoints = {
+            top: rect.top,
+            left: rect.left,
+            right: rect.left + rect.width,
+            bottom: rect.top + rect.height
+        };
+
+        return rectPoints;
+    }
+
+    // Returns whether two rectangles are intersected
+    function doesIntersect (rect1, rect2) {
+        var intersectLeftRight;
+        var intersectTopBottom;
+
+        var rect1Points = rectToPoints(rect1);
+        var rect2Points = rectToPoints(rect2);
+
+        if (rect1.width >= 0) {
+            if (rect2.width >= 0)
+                intersectLeftRight = !((rect1Points.right <= rect2Points.left) || (rect2Points.right <= rect1Points.left));
+            else
+                intersectLeftRight = !((rect1Points.right <= rect2Points.right) || (rect2Points.left <= rect1Points.left));
+        } else {
+            if (rect2.width >= 0)
+                intersectLeftRight = !((rect1Points.left <= rect2Points.left) || (rect2Points.right <= rect1Points.right));
+            else
+                intersectLeftRight = !((rect1Points.left <= rect2Points.right) || (rect2Points.left <= rect1Points.right));
+        }
+
+        if (rect1.height >= 0) {
+            if (rect2.height >= 0)
+                intersectTopBottom = !((rect1Points.bottom <= rect2Points.top) || (rect2Points.bottom <= rect1Points.top));
+            else
+                intersectTopBottom = !((rect1Points.bottom <= rect2Points.bottom) || (rect2Points.top <= rect1Points.top));
+        } else {
+            if (rect2.height >= 0)
+                intersectTopBottom = !((rect1Points.top <= rect2Points.top) || (rect2Points.bottom <= rect1Points.bottom));
+            else
+                intersectTopBottom = !((rect1Points.top <= rect2Points.bottom) || (rect2Points.top <= rect1Points.bottom));
+        }
+
+        return intersectLeftRight && intersectTopBottom;
+    }
+
+    // Returns intersection points of two rectangles
```

Comment:
Could you provide some further documentation?
- Does this only work for non-rotated rectangles?
- It appears that the function returns coordinates as left, right, top, and bottom. Could it be expressed as two points instead?